### PR TITLE
Copy net-new PDD Backend into codebase

### PR DESCRIPTION
**Pull request by nickgraz**
**No changes have been made to any file under `src/main/kotlin` or `src/test`**. These files are identical to their current state in `pdd-backend:main`. This code does not compile. The intention with this pull request is that the contents of the files can be largely ignored, since they are identical to already reviewed application code from `pdd-backend`.

### Package restructuring
Application code was copied over into `terraware-server` into a new package `com.terraformation.backend.documentproducer`. The package structure in `pdd-backend` was flattened into this new package. 
```
pdd-backend folder                                     || terraware-server folder
src/main/kotlin/com/terraformation/pdd/document        -> src/main/kotlin/com/terraformation/backend/documentproducer
src/main/kotlin/com/terraformation/pdd/document/api    -> src/main/kotlin/com/terraformation/backend/documentproducer/api
src/main/kotlin/com/terraformation/pdd/document/db     -> src/main/kotlin/com/terraformation/backend/documentproducer/db
src/main/kotlin/com/terraformation/pdd/document/model  -> src/main/kotlin/com/terraformation/backend/documentproducer/model
src/main/kotlin/com/terraformation/pdd/methodology/api -> src/main/kotlin/com/terraformation/backend/documentproducer/api
src/main/kotlin/com/terraformation/pdd/variable        -> src/main/kotlin/com/terraformation/backend/documentproducer
src/main/kotlin/com/terraformation/pdd/variable/api    -> src/main/kotlin/com/terraformation/backend/documentproducer/api
src/main/kotlin/com/terraformation/pdd/variable/db     -> src/main/kotlin/com/terraformation/backend/documentproducer/db
src/main/kotlin/com/terraformation/pdd/variable/model  -> src/main/kotlin/com/terraformation/backend/documentproducer/model

The one exception to this rule is the admin controller, since all 
admin controllers exist in the same folder in `terraware-server`

src/main/kotlin/com/terraformation/pdd/admin/AdminController.kt -> src/main/kotlin/com/terraformation/backend/admin/AdminDocumentsController.kt
```

All test code was flattened in the same way.

### Migrations
- All necessary migrations from `pdd-backend` were brought over and concatenated into a single migration file at [V280](src/main/resources/db/migration/0250/V280__DocumentProducerTables.sql). With the exception of the modifications to the `files` table at the top of the migration, these migrations are an exact copy from `pdd-backend`.

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2029?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2030** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2030?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2029** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2029?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2033** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2033?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 14 other dependent PRs ([#2064](https://github.com/terraware/terraware-server/pull/2064) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2064?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2065](https://github.com/terraware/terraware-server/pull/2065) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2065?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2068](https://github.com/terraware/terraware-server/pull/2068) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2068?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 11 others)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/admin/AdminDocumentController.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/admin/AdminDocumentController.kt:1]: The PR comment made me think the new files were all going to be under `com.terraformation.backend.documentproducer` but this one looks like it's going in the Terraware admin package. Intentional?

File: src/main/kotlin/com/terraformation/backend/documentproducer/DocumentUpgradeCalculator.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/documentproducer/DocumentUpgradeCalculator.kt]: Were the package names supposed to be updated per the PR description?

File: src/main/resources/db/migration/0250/V280__DocumentProducerFilesModifications.sql
Comment (sgrimm) [inline src/main/resources/db/migration/0250/V280__DocumentProducerFilesModifications.sql]: The `caption` column on `report_photos` should probably go away (with the values moved to this new column).

File: src/main/resources/db/migration/0250/V281__Variables.sql
Comment (sgrimm) [inline src/main/resources/db/migration/0250/V281__Variables.sql]: What would you think of concatenating all these migrations into one file? I don't think there's much benefit in having them separated out when we just care about getting to the right end state.

File: src/main/resources/db/migration/0250/V284__CreatePdds.sql
Comment (sgrimm) [inline src/main/resources/db/migration/0250/V284__CreatePdds.sql]: I suppose we can leave this out.

**Review by sgrimm — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/terraformation/backend/documentproducer/DocumentUpgradeCalculator.kt
Comment (nickgraz) [reply src/main/kotlin/com/terraformation/backend/documentproducer/DocumentUpgradeCalculator.kt]: Sorry, I will update the description, the files were moved into the new folder structure, but as mentioned at the top of the description, none of the contents have changed. All of the modifications are made in a subsequent PR.

File: src/main/resources/db/migration/0250/V281__Variables.sql
Comment (nickgraz) [reply src/main/resources/db/migration/0250/V281__Variables.sql]: Makes sense to me.

File: src/main/resources/db/migration/0250/V280__DocumentProducerFilesModifications.sql
Comment (nickgraz) [reply src/main/resources/db/migration/0250/V280__DocumentProducerFilesModifications.sql]: Is it OK to remove that in a subsequent PR? The intention with this PR is to make as few changes to the original code as possible. While this particular file is different than what is in `pdd-backend`, I fear making that change will require other code changes which are out of scope of this PR.

File: src/main/kotlin/com/terraformation/backend/admin/AdminDocumentController.kt
Comment (nickgraz) [reply src/main/kotlin/com/terraformation/backend/admin/AdminDocumentController.kt:1]: I will update the description, this is the only class that deviates from the pattern because so far all of our admin controllers reside in the same place.

File: src/main/resources/db/migration/0250/V280__DocumentProducerFilesModifications.sql
Comment (sgrimm) [reply src/main/resources/db/migration/0250/V280__DocumentProducerFilesModifications.sql]: Yes, sorry, didn't mean to imply it should be part of this PR, just something that occurred to me as I was looking at this.

Comment (nickgraz) [reply src/main/resources/db/migration/0250/V280__DocumentProducerFilesModifications.sql]: Got it, I'll make sure to add it to a subsequent PR

**Review by sgrimm — APPROVED** [review]
This seems fine to merge into the feature branch and iterate on.

**Review by nickgraz — COMMENTED** [review]
I can't seem to just merge this one through graphite on its own because CI is failing, and no option to override. Since I have run into issues with merging mid-stack branches outside of graphite, I am going to wait until #2030 is merged into this first since that PR fixes compilation and tests.

**Pull request merged**
